### PR TITLE
Simplify profile avatar setting logic

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details
Update to the logic introduced in https://github.com/Expensify/App/pull/7366 and https://github.com/Expensify/App/pull/7785. Simplifies things a bit. 

Note that inconsistent `Save` button coloring on web and desktop appears to be an [existing regression](https://github.com/Expensify/App/issues/7784#issuecomment-1042377952)

cc: @mananjadhav 

### Fixed Issues

N/A found while looking at this regression https://github.com/Expensify/App/issues/7784

### Tests
1. Log into a new account on New Expensify and navigate to the profile settings 
2. Click on the profile picture and upload a photo
3. Confirm the photo preview  updates immediately  
4. Click the save button and confirm the photo saves as expected 
5. Modify your user's first and last name and click save
6. Confirm the name is updated and that your profile photo is maintained 
7. Click on the profile picture and remove the photo 
8. Confirm the photo preview updates immediately to a default avatar
9. Click save and confirm your profile photo is successfully updated 

### QA Steps
Same as tests 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/16747822/154382682-701ffaf9-b4dd-41fc-bf3a-88c729693473.mov

#### Mobile Web

https://user-images.githubusercontent.com/16747822/154388783-ee674b00-7de0-4814-8ae9-e32d02b3c6d0.mov

#### Desktop

https://user-images.githubusercontent.com/16747822/154388797-0fd43b32-a2bd-4c0d-beb7-4c8cab1f3dcf.mov

#### iOS

https://user-images.githubusercontent.com/16747822/154386363-d942022d-6562-44e0-85bd-56a7f2e6cec0.mov

#### Android

Android device is super slow, will work on getting a video tomorrow

### CODE REVIEW

**Comment by mananjadhav** [comment]
Saw the changes @NikkiWines, yeah this simplified the logic a lot! Thanks for sharing this.

File: src/pages/settings/Profile/ProfilePage.js [hunk 1]
```diff
@@ -186,14 +169,17 @@ class ProfilePage extends Component {
             return;
         }
 
-        if (this.state.avatarImage) {
-            PersonalDetails.setAvatar(this.state.avatarImage);
-        }
+        // Check if the user has modified their avatar
+        if ((this.props.myPersonalDetails.avatar !== this.state.avatar.uri) && this.state.isAvatarChanged) {
+            // If the user removed their profile photo, replace it accoridngly with the default avatar
```
Comment (mananjadhav) [inline src/pages/settings/Profile/ProfilePage.js:174]:
```suggestion
            // If the user removed their profile photo, replace it accordingly with the default avatar
```

**Comment by timszot** [comment]
LGTM once the spelling is corrected.

File: src/pages/settings/Profile/ProfilePage.js (hunk 1 above)
Comment (NikkiWines) [reply src/pages/settings/Profile/ProfilePage.js:174]: oops 😄

**Comment by NikkiWines** [comment]
Updated!

**Review by timszot — APPROVED** [review]

**Comment by NikkiWines** [comment]
@SumitDiyora if possible could you make this a new issue for these bugs following [this template](https://github.com/Expensify/App/issues/new?assignees=&labels=&template=Accessibility.md) so that they can easily be picked up and worked on? Thank you!

**Pull request merged**